ui/src/message_popup/author/edit_question_page.jsx

```diff
@@ -0,0 +1,25 @@
+/* @flow weak */
+import React from 'react';
+import _ from 'lodash';
+
+import {allQuestions} from '../questions.js';
+import {withStudents, withIndicator} from '../transformations.jsx';
+
+import QuestionPage from './question_page.jsx';
+
+export default React.createClass({
+  displayName: 'EditQuestionPage',
+
+  propTypes: {
+    questionId: React.PropTypes.string.isRequired
+  },
+
+  render(){
+    const question = withIndicator(_.find(withStudents(allQuestions), question => question.id.toString()===this.props.questionId));
```

Comment:
Would it make sense to lookup the question first, then join in the student and indicator data?